### PR TITLE
Add ability to have local-only stories, Add scaffolding for PiPiP component to demonstrate

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What
* Have an `INTERNAL` folder in storybook for internal-only stories
  * These only display on development environments and not on production environements
* Ensure that communication-react is not exporting _internal exports from react-components
* Add empty PiPiP component to show story and ensure exports not added to API file

# Why
We will have non-exported pure components by nature of only being able to support a subset of everything we create. However as a local dev I need to be able to run and see the component in storybook for local development

# How Tested
* `start storybook` (development env) - can see the Internal folder and the stories in side 
* `build storybook` (production env) - Internal folder is not there

### CODE REVIEW

File: packages/communication-react/review/communication-react.api.md [hunk 1]
```diff
@@ -1270,32 +1270,6 @@ export interface HorizontalGalleryStyles extends BaseCustomStyles {
     previousButton?: IStyle;
 }
 
-// @internal
-export const _IdentifierProvider: (props: _IdentifierProviderProps) => JSX.Element;
-
-// @internal
-export interface _IdentifierProviderProps {
```
Comment (JamesBurnside) [inline packages/communication-react/review/communication-react.api.md:1277]: We actually exported this from our GA package

Comment (prprabhu-ms) [reply packages/communication-react/review/communication-react.api.md:1277]: Would recommend splitting this change out of this PR and discussing it internally. I think this was only partially "accidental".

File: packages/react-components/src/components/PictureInPictureInPicture/PictureInPictureInPicture.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import React from 'react';
+import { VideoTileProps } from '../VideoTile';
+
+/**
+ * Props for {@link _PictureInPictureInPicture} component.
+ *
+ * @internal
+ */
+export interface _PictureInPictureInPictureProps {
```
Comment (prprabhu-ms) [inline packages/react-components/src/components/PictureInPictureInPicture/PictureInPictureInPicture.tsx:12]: I'm assuming this props definition is also only for demonstration purposes.

**Review by prprabhu-ms — APPROVED** [review]

File: packages/communication-react/review/communication-react.api.md (hunk 1 above)
Comment (JamesBurnside) [reply packages/communication-react/review/communication-react.api.md:1277]: Great point, will do

File: packages/react-components/src/components/PictureInPictureInPicture/PictureInPictureInPicture.tsx (hunk 2 above)
Comment (JamesBurnside) [reply packages/react-components/src/components/PictureInPictureInPicture/PictureInPictureInPicture.tsx:12]: Yeah, it'd v. much subject to change depending on implementation - I'll simplify for this PR

Comment (JamesBurnside) [reply packages/react-components/src/components/PictureInPictureInPicture/PictureInPictureInPicture.tsx:11]: updated

File: packages/communication-react/review/communication-react.api.md (hunk 1 above)
Comment (JamesBurnside) [reply packages/communication-react/review/communication-react.api.md]: updated

**Review by anjulgarg — APPROVED** [review]

**Comment by anjulgarg** [comment]
Awesome. Any public component will need to be exported explicitly now from communication-react?

**Comment by JamesBurnside** [comment]
> Awesome. Any public component will need to be exported explicitly now from communication-react?

Yeah, the alternative would be to have two index.ts (so have `index.ts` and  have a `index.internal.ts`) and alias storybook to the internal one, but I like the explicit nature in the communication-react folder (as currently we have an `@internal` being exported -- presumably unintentionally??)

**Comment by prprabhu-ms** [comment]
I realized that chromatic links don't show internal stories. That's less than ideal. Any way we could get the links there as well?

**Comment by JamesBurnside** [comment]
> I realized that chromatic links don't show internal stories. That's less than ideal. Any way we could get the links there as well?

They actually do unintentionally which I'm fixing, for storybook builds we can set the env variable to allow it - I'll do a PR now fixing that

**Pull request merged**